prov/gni/src/gnix_ep.c

```diff
@@ -361,7 +361,7 @@ static inline int __gnix_ep_fini_vc(struct gnix_fid_ep *ep)
 		while ((vc = (struct gnix_vc *)
 				_gnix_vec_iterator_next(&iter))) {
 			_gnix_vec_remove_at(ep->vc_table,
-					    GNIX_VECTOR_ITERATOR_IDX(iter));
+					    GNIX_VECTOR_ITERATOR_IDX(iter) - 1);
```

Comment:
If you this route you should do the decrement in this macro instead:

 #define GNIX_VECTOR_ITERATOR_IDX(_iter)	((_iter).cur_idx - 1)